client/src/providers/Context.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useContext, createContext, useEffect } from "react";
+import axios from "axios";
+import PropTypes from "prop-types";
+
+// CREATE USER CONTEXT
+export const UserContext = createContext();
+
+//CUSTOM HOOKS
+// Get Auth
+export const useAuth = () => {
+  // Usage:
+  // const authenticated = useAuth();
+  const authenticated = useContext(UserContext).authenticated;
+  return authenticated;
+};
+// Set Auth
+export const useSetAuthenticated = () => {
+  // Usage:
+  // const setAuthenticated = useSetAuthenticated();
+  // setAuthenticated(true) OR setAuthenticated(false);
+  const setAuthenticated = useContext(UserContext).setAuthenticated;
+  return setAuthenticated;
+};
+// Get User Data
+export const useUserContext = () => {
+  return useContext(UserContext).userData;
+};
+// Set User Data has Updated
+export const useSetUserHasUpdated = () => {
```

Comment:
If I understand correctly with the set up, the child component calls a function which triggers the context to update on its own and renders the child with updated props. In that case, we should name this `useUpdateData` and have it return a function, which can be invoked without any argument.
```
const useUpdateData= ()=>{
    // Usage: 
    // const updateData = useUpdateData()
    // updateData()
    const setUserHasUpdated = useContext(UserContext).setUserHasUpdated;
    const updateData = () => {setUserHasUpdated(true)}
    return updateData
}
```

Reply:
It's must cleaner without needing to be passed a boolean argument in the child component. Thanks!